pyEpiabm/pyEpiabm/sweep/update_place_sweep.py

```diff
@@ -148,3 +157,93 @@ def update_place_group(self, place, mean_capacity: float = 25,
+            person_list = (place.cell.persons).copy()
+        # Ensure that the number of people put in the place
+        # is at most its capacity or the total number of
+        # people in the cell. Will use a power law calculation if
+        # parameters are provided, and a Poisson distribution is not.
+        if power_law_params is None:
+            new_capacity = np.random.poisson(mean_capacity)
+        else:
+            assert len(power_law_params) == 3, \
+                ("Incorrect number of power law parameters given"
+                 + " - should be of the form [maximum, offset, power]")
+            [maximum, offset, power] = power_law_params
+            s = (offset / (offset + maximum - 1)) ** power
+            r = random.random()
+            num = offset * ((1 - s) * r + s) ** (-1 / power) + 1 - offset
+            new_capacity = math.floor(num)
+
+        new_capacity = min(new_capacity, len(person_list))
+
+        if len(person_list) <= 0:
+            logging.info("No people in the person list supplied"
+                         + " to update " + str(place))
+            return
+        if person_weights == [0 for _ in person_list]:
+            logging.info("List of 0 weights given: no people"
+                         + " of acceptable age for this place")
+            return
+        count = 0
+
+        while count < new_capacity:
+            if person_weights is not None:
+                assert len(person_weights) == len(person_list),\
+                    ('Weights given is a different size to the person list.')
+                person = random.choices(person_list, person_weights, k=1)[0]
+            else:
+                i = random.randint(1, len(person_list))
+                person = person_list[i-1]
+            # Checks person is not already in the place, and that they
+            # haven't already been assigned to this place type.
+            if person.age >= carehome_params["carehome_minimum_age"]:
+                if ((person not in place.persons) and
+                        (place.place_type not in person.place_types)):
+                    place.add_person(person, 1)
+                    person.care_home_resident = 1
+                count += 1
+            elif person.age < carehome_params["carehome_minimum_age"]:
+                if ((person not in place.persons) and
+                        (place.place_type not in person.place_types)):
+                    place.add_person(person, 0)
+                    person.key_worker = 1
+                count += 1
+            # Prevent person being readded to list
+            if person_weights is not None:
+                person_weights.pop(person_list.index(person))
+            person_list.remove(person)
+
+            if len(person_list) <= 0:
+                # logging.warning("Insufficient people in the person list"
+                #                 + " supplied to update " + str(place))
+                break
```

Comment:
1. 'mean_capacity' tend to be quite different for residents and workers. I suggest to differ them if possible.
2. group_index and group_size seem not to be used in actual coding below so it's possible to remove these parameters.
3. This block seems to be quite similar to the function update_place_group, could it be merged a bit? (e.g. make line 229 to 239 a small function, check carehome by if statement and call it small function directly)

Reply:
I think I've got a plan to bring the two functions together, so will need to keep the group_index and group_size parameters. The mean capacity refers to the capacity of the carehome overall (residents + workers).